fedimint-server/src/net/peers.rs

```diff
@@ -219,18 +181,16 @@ where
         delay_calculator: DelayCalculator,
         connect: PeerConnector<T>,
         task_group: &TaskGroup,
-    ) -> (Self, PeerStatusChannels) {
+        status_channels: Arc<RwLock<BTreeMap<PeerId, PeerConnectionStatus>>>,
```

Comment:
I think I liked it returning multiple values instead of taking extra arg more.

But even more so, having a shorter named newtype was nicer.

Reply:
BTW. Can't we just put `status_channels` in `self`?

Reply:
Though if I see correctly, you're getting rid of these channels there, and replacing everything with just shared variables, with is :+1: and the original channels there in a map seemed clunky indeed.

Reply:
It is actually not primarily about turning the return value into an argument but instead of being able to create the status_channels before we create the p2p network layer such that we can create the network layer after we build the consensus api which also contains a clone of status_channels. 

This allows us to more consistently factor fedimint server. Right now everything inside ConsensusServer::run is factored very consistently, in particular values are created as late as possible, right before we need them, so understanding a code section requires less context.

Reply:
I see. Makes sense.